host-controller/src/main/java/org/jboss/as/host/controller/discovery/DiscoveryOption.java

```diff
@@ -52,6 +52,7 @@
 
     /**
      *  Gets the host name of the remote domain controller.
+     *  This method may return null.
```

Comment:
It's messy. It can return null until the discover() method is invoked. I commented more on WFCORE-565.

I wanted to play with this yesterday so I could see what the read-resource and read-resource-description responses looked like, including with custom-discovery set up. I found a lot of problems, but not due to this patch.

https://github.com/bstansberry/wildfly-core/commits/WFCORE-565 had some quick changes just to get things so I could do my experiments.